cf-check/diagnose.c

```diff
@@ -507,6 +507,23 @@ static int fork_and_diagnose(const char *path, bool validate, bool test_write)
     return CF_CHECK_OK;
 }
 
+static char *follow_symlink(const char *path)
```

Comment:
Why not `realpath()` in the end?

Reply:
realpath doesn't work on broken symlinks :(